warringBishops.js

```diff
@@ -0,0 +1,96 @@
+/*
+Imagine that we have an 8 x 8 chess board that has some bishops on it, and no other pieces. 
+In chess, bishops can travel diagonally in all 4 directions, and they can move as far as they want in these directions 
+(however, if there is another piece in the way, they cannot move past that piece).
+
+You'll be a given an array of arrays. Each of the smaller arrays has size 2, and represents 1 bishop. 
+For example, if we have, [[1,2]], then there is a bishop at row = 1, col = 2.
+
+If 2 bishops can attack each other (i.e. they are in the same diagonal), we'll say that these are warring bishops. 
+Additionally, if bishop 1 can attack bishop 2, and bishop 2 can attack bishop 3, then all 3 of these are warring bishops.
+
+Given the input list of bishops, return true if all the bishops are part of the same group of warring bishops. Otherwise, return false.
+*/
+
+const { Queue } = require('./utils/queue');
+
+const areWarring = (bishopOne, bishopTwo) => {
+    const [bishopOneRow, bishopOneCol] = bishopOne;
+    const [bishopTwoRow, bishopTwoCol] = bishopTwo;
+
+    const slope = Math.abs((bishopTwoCol - bishopOneCol) / (bishopTwoRow - bishopOneRow));
```

Comment:
There's a slight error here. The formula for slope is (delta y) / (delta x). This can equivalently be written as (y2 - y1) / (x2 - x1).

In fact, rows correspond with y-values and cols correspond with x-values.

Imagine a 2D matrix, and then imagine picking some position and then increasing your col value by one. Which direction are you stepping in? Does that correspond to a change in your x-coordinate or a change in your y-coordinate?